packages/iov-core/src/signingservercore.ts

```diff
@@ -49,10 +77,12 @@ export class SigningServerCore {
         throw new Error("More than one wallets contain the identity to sign this transaction");
     }
 
-    // TODO: ask user for permission
-
-    const response = await this.signer.signAndPost(transaction, walletId);
-    return response.transactionId;
+    if (this.authorizeSignAndPost(reason, transaction)) {
+      const response = await this.signer.signAndPost(transaction, walletId);
+      return response.transactionId;
+    } else {
+      return undefined;
```

Comment:
I prefer promise rejection on error, rather than return undefined.

DIfference between `Maybe<TransactionId>` and `Result<TransactionId, Error>` in functional languages

Reply:
This is not an error but a decision made by the user, which must have one explicit handleable result that is transmitted via a JSON-RPC response.

Will improve documentation of this function as well